manifests/init.pp

```diff
@@ -73,7 +73,7 @@
   Boolean $generate = true,
   Boolean $regenerate = false,
   Boolean $deploy = true,
-  String $ca_common_name = $certs::params::ca_common_name,
+  Stdlib::Fqdn $ca_common_name = $certs::params::ca_common_name,
```

Comment:
Given this is the CA's common name, the previous string requirement should remain, as you should be able to put whatever descriptor you want for the CA here. Take puppet for example:

```
        Subject: CN=Puppet CA: pipe-katello-server-nightly-centos7.woodford.example.com
```